src/runtime/tests/refiner-test.ts

```diff
@@ -577,64 +610,128 @@ describe('KTExtracter', () => {
     typeFor: (type: string) => type === 'Number' ? 'Double' : type,
     defaultValFor: (type: string) => ({'Number': '0.0', 'Boolean': 'false', 'Text': '""'}[type])
   };
-  it('tests can create queries from refinement expressions involving math expressions', Flags.withFieldRefinementsAllowed(async () => {
+  it('creates queries from refinement expressions involving math expressions', Flags.withFieldRefinementsAllowed(async () => {
+      const manifest = await Manifest.parse(`
+        particle Foo
+          input: reads Something {a: Number [ a > ? and a != 100 ], b: Number [b > 20 and b < 100] } [a + b/3 > 100]
+      `);
+      const schema = manifest.particles[0].handleConnectionMap.get('input').type.getEntitySchema();
+      const query: string = KTExtracter.queryFromSchema(schema, escaper);
+      assert.strictEqual(query,
+        `\
+val a = data.singletons["a"].toPrimitiveValue(Double::class, 0.0)
```

Comment:
These tests are helpful for seeing the behavior.

Reply:
Thanks, thought it'd be good to have a few more fine grained tests than just the goldens.